microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/internal/core/providers/MicroProfileFaultToleranceProvider.java

```diff
@@ -0,0 +1,328 @@
+import org.eclipse.jdt.core.JavaModelException;
+
+import com.redhat.microprofile.commons.DocumentFormat;
+import com.redhat.microprofile.commons.MicroProfilePropertiesScope;
+import com.redhat.microprofile.jdt.core.AbstractAnnotationTypeReferencePropertiesProvider;
+import com.redhat.microprofile.jdt.core.IPropertiesCollector;
+import com.redhat.microprofile.jdt.core.SearchContext;
+import com.redhat.microprofile.jdt.core.utils.IJDTUtils;
+
+/**
+ * Properties provider to collect MicroProfile properties from the MicroProfile
+ * Fault Tolerance annotations.
+ * 
+ * @author Angelo ZERR
+ * 
+ * @see https://github.com/eclipse/microprofile-fault-tolerance/blob/master/spec/src/main/asciidoc/configuration.asciidoc
+ *
+ */
+public class MicroProfileFaultToleranceProvider extends AbstractAnnotationTypeReferencePropertiesProvider {
+
+	private static final String MICROPROFILE_FAULT_TOLERANCE_CONTEXT_KEY = MicroProfileFaultToleranceProvider.class
+			.getName() + "#MicroProfileFaultToleranceContext";
+	private static final String ASYNCHRONOUS_ANNOTATION = "org.eclipse.microprofile.faulttolerance.Asynchronous";
+
+	private static final String BULKHEAD_ANNOTATION = "org.eclipse.microprofile.faulttolerance.Bulkhead";
+
+	private static final String CIRCUITBREAKER_ANNOTATION = "org.eclipse.microprofile.faulttolerance.CircuitBreaker";
+
+	private static final String FALLBACK_ANNOTATION = "org.eclipse.microprofile.faulttolerance.Fallback";
+
+	private static final String RETRY_ANNOTATION = "org.eclipse.microprofile.faulttolerance.Retry";
+
+	private static final String TIMEOUT_ANNOTATION = "org.eclipse.microprofile.faulttolerance.Timeout";
+
+	private static final String[] ANNOTATION_NAMES = { ASYNCHRONOUS_ANNOTATION, BULKHEAD_ANNOTATION,
+			CIRCUITBREAKER_ANNOTATION, FALLBACK_ANNOTATION, RETRY_ANNOTATION, TIMEOUT_ANNOTATION };
+
+	@Override
+	protected String[] getAnnotationNames() {
+		return ANNOTATION_NAMES;
+	}
+
+	static class AnnotationInfo {
+
+		private final String name;
+
+		private final String simpleName;
+
+		private final List<AnnotationParameter> parameters;
+
+		public AnnotationInfo(IType annotation, IJDTUtils utils, DocumentFormat documentFormat)
+				throws JavaModelException {
+			this.name = annotation.getFullyQualifiedName();
+			this.simpleName = annotation.getElementName();
+			this.parameters = new ArrayList<>();
+			IMethod[] methods = annotation.getMethods();
+			if (methods != null) {
+				for (IMethod method : methods) {
+
+					// name
+					String name = method.getElementName();
+
+					// type
+					String methodResultTypeName = getResolvedResultTypeName(method);
+					IType returnType = findType(method.getJavaProject(), methodResultTypeName);
+					String type = getPropertyType(returnType, methodResultTypeName);
+
+					try {
+						utils.discoverSource(method.getClassFile(), new NullProgressMonitor());
+					} catch (CoreException e) {
+						// TODO Auto-generated catch block
+						e.printStackTrace();

```

Comment:
use logger / send to client

Reply:
@fbricon code has been changed, and now it's fixed.